internal/repository/index.go

```diff
@@ -26,27 +28,72 @@ type Index struct {
 }
 
 type indexEntry struct {
-	packID restic.ID
-	offset uint
-	length uint
+	packID *restic.ID
+	offset uint32
+	length uint32
 }
 
 // NewIndex returns a new index.
 func NewIndex() *Index {
 	return &Index{
-		pack:    make(map[restic.BlobHandle][]indexEntry),
-		created: time.Now(),
+		blob:       make(map[restic.BlobHandle]indexEntry),
+		duplicates: make(map[restic.BlobHandle][]indexEntry),
+		created:    time.Now(),
 	}
 }
 
-func (idx *Index) store(blob restic.PackedBlob) {
+// allBlobs return all blobs to the given blobHandle saved in the index
+// also returns if there exists a blob to the handle or not.
```

Comment:
Why the second return value? The caller can just check whether the length of the first return value is 0.

Reply:
You are right, thanks for the hint. This is changed now!